### PR TITLE
Adjust publish data to new API

### PR DESCRIPTION
**Pull request by netoax**
**Describe what this PR introduces:** 

This PR updates the sending of `data.published` to send authorization token and rename the network message structure from `DataPublish` to `DataSent`.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [ ] Feature
- [ ] Code style update
- [X] Refactor
- [ ] Build-related changes
- [X] Other, please describe: adjustments to new API

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [X] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform: macOS Darwin 18.0.0
- Go version: 1.14

If you have relevant logs, error output and etc, please attach them.

**Other information:**

### CODE REVIEW

File: pkg/thing/delivery/amqp/client.go [hunk 1]
```diff
@@ -144,18 +144,21 @@ func (cs *commandSender) SendListResponse(things []*entities.Thing, corrID strin
...
 	}
 
-	return cs.amqp.PublishPersistentMessage(exchangeDevices, exchangeDevicesType, corrID, msg)
+	return cs.amqp.PublishPersistentMessage(exchangeDevices, exchangeDevicesType, corrID, msg, nil)
 }
 
 // PublishUpdateData send update data command
-func (mp *msgClientPublisher) PublishPublishedData(thingID string, data []entities.Data) error {
-	resp := &network.DataUpdate{ID: thingID, Data: data}
+func (mp *msgClientPublisher) PublishPublishedData(thingID, token string, data []entities.Data) error {
```
Comment (lcbm) [inline pkg/thing/delivery/amqp/client.go]: This comment doesn't match the method's name... could you update it? 😄

**Review by lcbm — APPROVED** [review]
I'm approving this in advance, but I made a single remark in this PR -- make sure you take a look at it before rebasing and merging 😄 💯

File: pkg/thing/delivery/amqp/client.go (hunk 1 above)
Comment (netoax) [reply pkg/thing/delivery/amqp/client.go]: Done, `vscode` didn't warn me and `golangci-lint` disable it by default. In another PR, I'll enforce this rule and fix some errors. Thanks!

**Review by esdrasjnr — APPROVED** [review]
LGTM!

**Review by esdrasjnr — APPROVED** [review]
Sorry, I didn't approve in last comment :sweat_smile:

**Pull request merged**